src/components/Form/FormFields/PhoneNumberFormField.tsx

```diff
@@ -17,141 +17,151 @@ import {
+    };
+
+    useEffect(() => {
+      if (field.value && field.value.length > 0) {
+        if (field.value.startsWith("1800")) {
+          setCountry({ flag: "📞", name: "Support", code: "1800" });
+          return;
+        }
+        if (field.value === "+") {
+          setCountry({ flag: "🌍", name: "Other", code: "+" });
+          return;
+        }
+        setCountry(phoneCodes[getCountryCode(field.value)!]);
       }
-      setCountry(phoneCodes[getCountryCode(field.value)!]);
-    }
-  }, [setValue]);
+    }, [setValue]);
 
-  return (
-    <FormField
-      field={{
-        ...field,
-        error: field.error ?? error,
-        labelSuffix: field.labelSuffix || (
-          <PhoneNumberTypesHelp types={props.types} />
-        ),
-      }}
-    >
-      <div className="relative rounded-md shadow-sm">
-        <Popover>
-          {({ open }: { open: boolean }) => {
-            return (
-              <>
-                <PopoverButton className="absolute h-full">
-                  <div className="absolute inset-y-0 left-0 m-0.5 flex w-[4.5rem] cursor-pointer items-center justify-around bg-slate-100">
-                    <span className="rounded-md pl-4">
-                      {country?.flag ?? "🇮🇳"}
-                    </span>
-                    <CareIcon
-                      icon="l-angle-down"
-                      className={`text-2xl font-bold ${open && "rotate-180"}`}
-                    />
-                  </div>
-                </PopoverButton>
-                <input
-                  type="tel"
-                  id={field.id}
-                  name={field.name}
-                  autoComplete={props.autoComplete ?? "tel"}
-                  className={classNames(
-                    "cui-input-base h-full pl-20 tracking-widest sm:leading-6",
-                    field.error && "border-danger-500",
-                    field.className,
-                  )}
-                  maxLength={field.value?.startsWith("1800") ? 11 : 15}
-                  placeholder={props.placeholder}
-                  value={formatPhoneNumber(field.value, props.types)}
-                  onChange={(e) => setValue(e.target.value)}
-                  disabled={field.disabled}
-                  onBlur={() => setError(validate(field.value, "blur"))}
-                />
-                <PopoverPanel className="w-full">
-                  {({ close }) => (
-                    <CountryCodesList
-                      handleCountryChange={handleCountryChange}
-                      onClose={close}
-                    />
-                  )}
-                </PopoverPanel>
-              </>
-            );
-          }}
-        </Popover>
-      </div>
-    </FormField>
-  );
-}
+    return (
+      <FormField
+        field={{
+          ...field,
 
+          error: field.error || error,
+          labelSuffix:
+            field.labelSuffix ||
+            (props.phoneNumberHelp == undefined && (
```

Comment:
shouldn't this be the opposite based on how it's read?

also keep the name `showHelp`. It's pretty obvious it's phone number help in this context as the prop is part of the phone number component.

```suggestion
            (props.phoneNumberHelp && (
```